CODING-GUIDELINES.md

```diff
@@ -0,0 +1,87 @@
+# Coding Guidelines
```

Comment:
I would rename this file to `CODING_GUIDELINES.md` to be consistent with other .md files

Reply:
OK